### ISSUE TITLE
False positive on C404

### ISSUE THREAD

**Issue by schinckel** [adamchainz/flake8-comprehensions#194]
I have a block of code that looks like:

    instance_kwargs = dict([
        key_value.split('=')
        key_value in full_lookup.split(':')[1].split('&')
    ])

This is being marked as  C404: unnecessary list comprehension.

However, I don't believe it's possible to use a single dict comprehension to perform this.

(I'm not saying it's great code, btw...)

**Comment by adamchainz** [comment]
Thanks. You should be able to use a generator to get around this for the time being

```python
In [1]: x = 'a=1;b=2'

In [2]: dict([i.split('=') for i in x.split(';')])
Out[2]: {'a': '1', 'b': '2'}

In [3]: dict((i.split('=') for i in x.split(';')))
Out[3]: {'a': '1', 'b': '2'}
```

**Comment by adamchainz** [comment]
Released in 3.1.0: https://pypi.org/project/flake8-comprehensions/3.1.0/
